config/crds/tenancy.kcp.dev_clusterworkspacetypes.yaml

```diff
@@ -44,6 +54,37 @@ spec:
                 description: additionalWorkspaceLabels are a set of labels that will
                   be added to a ClusterWorkspace on creation.
                 type: object
+              allowedChildWorkspaceTypes:
+                default: []
+                description: "allowedChildWorkspaceTypes is a list of ClusterWorkspaceTypes
+                  that can be created in a workspace of this type. \n By default,
+                  no type is allowed. This means no other workspace can be nested
+                  within a workspace of the given type. The name `*` allows any child
+                  type to be nested."
+                items:
+                  description: ClusterWorkspaceTypeName is a name of a ClusterWorkspaceType
+                  pattern: ^[A-Z][a-zA-Z0-9]+$
+                  type: string
+                type: array
+              allowedParentWorkspaceTypes:
+                description: "allowedParentWorkspaceTypes is a list of ClusterWorkspaceTypes
+                  that this type can be created in. \n By default, no type is allowed.
+                  This means no other workspace can have a workspace of the given
+                  type nested inside it. The name `*` allows any parent type to nest
+                  this one."
+                items:
+                  description: ClusterWorkspaceTypeName is a name of a ClusterWorkspaceType
+                  pattern: ^[A-Z][a-zA-Z0-9]+$
+                  type: string
```

Comment:
must be fully qualified

Reply:
same

Reply:
Follow-up.

Reply:
moved as follow-up into https://github.com/kcp-dev/kcp/issues/1149